prestoadmin/util/version_util.py

```diff
@@ -0,0 +1,189 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Stuff to handle version ranges.
+"""
+
+import re
+
+TD_VERSION = re.compile(r'^\d+t$')
+ANCIENT_TAGGED_VERSION = re.compile(r'^\d+t?-[A-Z]+$')
+
+
+def split_version(version_string):
+    return version_string.strip().split('.')
+
+
+def strip_tag(version):
+    """
+    Strip trailing non-numeric components from a version leaving the Teradata
+    't' on the final version component if it's present.
+    ['1', '2', 'THREE'] -> (1, 2)
+    ['1', 'TWO', '3'] -> raises a ValueError
+    ['0', '115t', 'SNAPSHOT'] -> (0, '115t')
+    ['ZERO', '123t'] -> raises a ValueError
+    :param version: something that can be sliced
+    :return: a tuple containing only integer components, except for possibly
+             the last one, which will be a string iff it's an integer followed
+             by the letter 't'
+    """
+    is_teradata = False
+    is_ancient = False
+
+    result = list(version[:])
+    while True:
```

Comment:
I still think a regex would be clearer (because it helps you understand what the version string is supposed to be), but I see the value of longer version string potential. maybe we could compromise and allow for there to be another value (e.g. 0.115t.1) but only 1? I don't think our version strings would get more nested than that.